src/components/map/WebViewMap.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react-native/no-color-literals */
+import React, { useCallback } from 'react';
+import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
+import { MapMarker, WebViewLeaflet, WebviewLeafletMessage } from 'react-native-webview-leaflet';
+import { colors } from '../../config';
+import { location as locationMarker } from '../../icons';
+
+type Props = {
+  locations: MapMarker[];
+  mapCenterPosition?: {
+    lat: number;
+    lng: number;
+  };
+  onMessageReceived?: (message: WebviewLeafletMessage) => void;
+  style?: StyleProp<ViewStyle>;
+  zoom?: number;
+};
+
+export const WebViewMap = ({
+  locations,
+  mapCenterPosition,
+  onMessageReceived,
+  style,
+  zoom
+}: Props) => {
+  const messageHandler = useCallback(
+    (message) => {
+      onMessageReceived?.(message);
+    },
+    [onMessageReceived]
+  );
+
+  return (
+    <View style={[styles.map, style]}>
+      <WebViewLeaflet
+        backgroundColor={'gray'}
+        onMessageReceived={messageHandler}
+        mapLayers={[
+          {
+            attribution: '© <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+            baseLayerIsChecked: true,
+            baseLayerName: 'OpenStreetMap.Mapnik',
+            url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
+          }
+        ]}
+        mapMarkers={[
+          ...locations.map((location) => {
+            return {
+              ...{
+                icon: locationMarker(colors.primary),
+                iconAnchor: { x: 9, y: 20 }
+              },
+              ...location
+            };
+          })
+        ]}
+        mapCenterPosition={mapCenterPosition ?? locations?.[0]?.position}
+        zoom={zoom}
+      />
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  map: {
+    aspectRatio: 16 / 9,
```

Comment:
:o nice! never used it. maybe that is the solution to go with also for ours images instead of calculating on our own?

Reply:
but this does not work for landscape. the map fills everything and the screen cannot be handled anymore.

![Screenshot_20201208-195432](https://user-images.githubusercontent.com/1942953/101528897-22643b00-3990-11eb-85b8-b95e2e5f8167.png)

you can see in the testflight version for bad belzig for example, that images always stay the same size on landscape and portrait. the imagecarousel changes to a version previews to the left and right of the main image.

![Screenshot_20201208-200328](https://user-images.githubusercontent.com/1942953/101529316-b3d3ad00-3990-11eb-98d6-59cd67c904a9.png)

the map should also not increase in size, or not that much. maybe this would work if the layout change on orientation change would apply in this branch. could be that the 100% width depend on the smaller content area and the map gets smaller. needs to be checked.

Reply:
the map is wrapped by a `WrapperWithOrientation` that has lesser width.

the issue lies with the orientation not being updated properly on android. (see above)